framework/wazuh/core/tests/test_wazuh_socket.py

```diff
@@ -2,42 +2,33 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
-from unittest.mock import patch, MagicMock, AsyncMock
+from unittest.mock import patch, MagicMock, call
+from asyncio import BaseEventLoop, BaseProtocol, StreamWriter, StreamReader, BaseTransport
+from struct import pack
 
 import pytest
 from wazuh.core.exception import WazuhException
-from wazuh.core.wazuh_socket import WazuhSocket, WazuhSocketJSON, WazuhAsyncSocket, WazuhAsyncProtocol, \
-    SOCKET_COMMUNICATION_PROTOCOL_VERSION, create_wazuh_socket_message
+from wazuh.core.wazuh_socket import WazuhSocket, WazuhSocketJSON, \
+     SOCKET_COMMUNICATION_PROTOCOL_VERSION, create_wazuh_socket_message, WazuhAsyncSocket, \
+     WazuhAsyncSocketJSON
+
+@pytest.fixture
+def oux_conn_patch():
+    """Fixture with asyncio.open_unix_connection patched."""
+    return patch('asyncio.open_unix_connection',
+                 return_value=(StreamReader(),StreamWriter(protocol=BaseProtocol(),
+                                                           transport=BaseTransport(),
+                                                           loop=BaseEventLoop(),
+                                                           reader=None)))
 
-
-class LoopMock:
-
-    def __init__(self):
-        pass
-
-    @staticmethod
-    async def create_connection(protocol_factory, sock):
-        with patch('asyncio.transports.Transport') as transport:
-            return transport, WazuhAsyncProtocol(LoopMock())
-
-    def set_exception_handler(self, exc_handler):
-        pass
-
-    def create_future(self):
-        pass
-
-
-class ProtocolMock:
-        def __init__(self):
-            self.on_data_received = ProtocolMock.testing()
-
-        @staticmethod
-        async def testing():
-            return 'test'
-        
-        @staticmethod
-        def get_data():
-            return b'data'
+@pytest.mark.asyncio
```

Comment:
```suggestion

@pytest.mark.asyncio
```

Reply:
Fixed in commit 78e38c8d101c7ebce4a5b6820400cdcd54c2efd9 and documented results of tests in https://github.com/wazuh/wazuh/issues/23839#issuecomment-2161522591.